src/flask_ml/flask_ml_server/MLServer.py

```diff
@@ -58,66 +59,190 @@ def __init__(self, name):
-                    "schema": schema,
-                }
-                routes.append(route_info)
-            return jsonify(routes)
-
-    def route(self, rule: str, input_type: str):
+                for endpoint in self.endpoints
+            ]
+            return jsonify(APIRoutes(root=routes).model_dump(mode="json"))
+
+    def route(
+        self,
+        rule: str,
+        task_schema_func: Optional[Callable[[], TaskSchema]] = None,
+        short_title: Optional[str] = None,
+        order: int = 0,
+    ):
         """
         rule : str - the name of the endpoint
         input_type : str - the type of the input data
         """
-        if rule is None:
-            raise ValueError('The parameter "rule" cannot be None')
-        if input_type is None:
-            raise ValueError('The parameter "input_type" cannot be None')
-        if type(rule) != str:
-            raise ValueError('The parameter "rule" is expected to be a string')
-
-        def build_route(ml_function):
-            self.endpoint2function[rule] = ml_function
-
-            @self.app.route(rule, endpoint=ml_function.__name__, methods=["POST"])
-            def wrapper():
-                try:
-                    data = request.get_json()
-                    data = RequestModel(**data)
-                    result = ml_function(data.inputs, data.parameters)
-                    response = ResponseModel(status="SUCCESS", results=result)
-                    return response.get_response()
-                except ValidationError as e:
-                    error_details = e.errors()
-                    error_details = [
-                        {
-                            "type": err.get("type", ""),
-                            "input": err.get("input", ""),
-                            "msg": err.get("msg", ""),
-                        }
-                        for err in error_details
-                    ]
-                    return ErrorResponseModel(
-                        status="VALIDATION_ERROR", errors=error_details
-                    ).get_response()
-
-            return wrapper
+
+        def build_route(ml_function: Callable[[Any, Any], ResponseBody]):
+            ensure_ml_func_parameters_are_typed_dict(ml_function)
+            if task_schema_func is not None:
+                ensure_ml_func_hinting_and_task_schemas_are_valid(ml_function, task_schema_func())
+                endpoint = EndpointDetails(
+                    rule=rule,
+                    task_schema_rule=rule + "/task_schema",
+                    sample_payload_rule=rule + "/sample_payload",
+                    payload_schema_rule=rule + "/payload_schema",
+                    func=ml_function,
+                    task_schema_func=task_schema_func,
+                    short_title=short_title or "",
+                    order=order,
+                )
+                self.endpoints.append(endpoint)

```

Comment:
Lots of repeated code. A lot of this can be eliminated. You can dynamically build dictionaries of parameters and pass the parameters by unpacking the dictionary.

```
di = {}
if event1: 
   di['param1'] = 1
   di['param2' ] = 2
else:
   di['param3'] = 4

common_func(**di)
```